DearPyGui/src/core/Theming/mvTheme.cpp

```diff
@@ -5,13 +5,42 @@ namespace Marvel {
 
 	mvTheme::mvTheme()
 	{
-		for (int i = 1; i < (int)mvAppItemType::ItemTypeCount; i++)
-			m_themeItems[(mvAppItemType)i] = mvAppItemTheme(i);
+		mvEventBus::Subscribe(this, SID("color_change"), mvEVT_CATEGORY_THEMES);
 	}
 
-	mvAppItemTheme* mvTheme::getThemeItem(mvAppItemType type)
+	mvTheme::~mvTheme()
 	{
-		return &m_themeItems.at(type);
+		mvEventBus::UnSubscribe(this);
+	}
+ 
+	bool mvTheme::onEvent(mvEvent& event)
+	{
+		mvEventDispatcher dispatcher(event);
+		dispatcher.dispatch(BIND_EVENT_METH(mvTheme::add_color), SID("color_change"), mvEVT_CATEGORY_THEMES);
+		return event.handled;
+	};
+
+	bool mvTheme::add_color(mvEvent& event)
+	{
+		mvAppItemType type = (mvAppItemType)(GetEInt(event, "ID") / 100);
+		int libID = GetEInt(event, "ID") % 100;
+		mvColor color = GetEColor(event, "COLOR");
+		std::string widget = GetEString(event, "WIDGET");
```

Comment:
Should get back a const reference. No need to copy the string.

Reply:
done